Core/src/org/sleuthkit/autopsy/modules/filetypeid/FileTypeDetector.java

```diff
@@ -474,7 +483,30 @@ private String detectAutopsyDefinedType(AbstractFile file) throws TskCoreExcepti
      */
     @Deprecated
     public String detectAndPostToBlackboard(AbstractFile file) throws TskCoreException {
-        return getFileType(file);
+        return getFileType(file, true);
+    }
+    
+    /**
+     * Gets the MIME type of a file, detecting it if it is not already known. If
+     * detection is necessary, the result is added to the case database.
+     *
+     * IMPORTANT: This method should only be called by ingest modules. All other
+     * clients should call AbstractFile.getMIMEType, and may call
+     * FileTypeDetector.detect, if AbstractFile.getMIMEType returns null.
+     *
+     * @param file The file.
+     *
+     * @return A MIME type name. If file type could not be detected or results
+     *         were uncertain, octet-stream is returned.
+     *
+     * @throws TskCoreException if detection is required and there is a problem
+     *                          writing the result to the case database.
+     * 
+     * @deprecated
+     */
+    @Deprecated
+    public String getFileType(AbstractFile file) throws TskCoreException {
+        return detect(file, true);
```

Comment:
I would like us to handle the public API change a bit differently.  Instead of creating a new API with a flag, let's call the new API detectFileType(AbstractFile) and have it call detect(file, false). The module writers do not need to know about our deferred write mechanism.